apps/homepage/src/app/[lang]/page.tsx

```diff
@@ -1,13 +1,87 @@
 'use client'
 import styles from './home.module.css'
-import { useLocalization } from '../../lib/context/loc-context'
+import { motion, AnimatePresence } from 'framer-motion';
+import { useEffect, useState } from 'react';
+import { UiViewport } from '@uireact/foundation';
+import { renderDynamicPhrase } from './utils';
+
+const phraseData = {
+  phrase: "We {keyWords} the future while being {keyDescriptions}",
+  keyWords: ['{engineer}', '{create}', '{code}'],
+  keyDescriptions: ['{humans}', '{developers}', '{artists}']
+}
 
 export default function AppPage() {
-  const loc = useLocalization();
+  const { getPhrase, next } = renderDynamicPhrase(phraseData);
+  const [phrase, setPhrase] = useState(getPhrase());
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setPhrase(next());
+    }, 3000);
+    return () => clearInterval(interval);
+  }, [next]);
+
+  const splitPhrase = (phrase: string) => {
+    return phrase.split(" ").map((word, idx) => ({
+      content: word,
+      isDynamic: word.startsWith("{") && word.endsWith("}"),
+      key: idx
+    }));
+  };
+
+  const phraseParts = splitPhrase(phrase);
 
   return (
-    <div className={styles.main}>
-      <h1 data-testid="title" className={styles.hero}>{loc.title}</h1>
-    </div>
+    <>
+      <UiViewport data-testid='hero message' criteria='l|xl'>
```

Comment:
Looks like the only difference between large and small is a class? `isMobile` instead of doing this I would add a media query to the stylesheet for it.